src/ORM/Paginator.php

```diff
@@ -0,0 +1,455 @@
+     */
+    public function validateSort(RepositoryInterface $object, array $options)
+    {
+        if (isset($options['sort'])) {
+            $direction = null;
+            if (isset($options['direction'])) {
+                $direction = strtolower($options['direction']);
+            }
+            if (!in_array($direction, ['asc', 'desc'])) {
+                $direction = 'asc';
+            }
+            $options['order'] = [$options['sort'] => $direction];
+        }
+        unset($options['sort'], $options['direction']);
+
+        if (empty($options['order'])) {
+            $options['order'] = [];
+        }
+        if (!is_array($options['order'])) {
+            return $options;
+        }
+
+        $inWhitelist = false;
+        if (isset($options['sortWhitelist'])) {
+            $field = key($options['order']);
+            $inWhitelist = in_array($field, $options['sortWhitelist'], true);
+            if (!$inWhitelist) {
+                $options['order'] = [];
+
+                return $options;
+            }
+        }
+
+        $options['order'] = $this->_prefix($object, $options['order'], $inWhitelist);
+
+        return $options;
+    }
+
+    /**
+     * Prefixes the field with the table alias if possible.
+     *
+     * @param \Cake\Datasource\RepositoryInterface $object Repository object.
+     * @param array $order Order array.
+     * @param bool $whitelisted Whether or not the field was whitelisted
+     * @return array Final order array.
+     */
+    protected function _prefix(RepositoryInterface $object, $order, $whitelisted = false)
+    {
+        $tableAlias = $object->alias();
+        $tableOrder = [];
+        foreach ($order as $key => $value) {
+            if (is_numeric($key)) {
+                $tableOrder[] = $value;
+                continue;
+            }
+            $field = $key;
+            $alias = $tableAlias;
+
+            if (strpos($key, '.') !== false) {
+                list($alias, $field) = explode('.', $key);
+            }
+            $correctAlias = ($tableAlias === $alias);
+
+            if ($correctAlias && $whitelisted) {
+                // Disambiguate fields in schema. As id is quite common.
+                if ($object->hasField($field)) {
+                    $field = $alias . '.' . $field;
+                }
+                $tableOrder[$field] = $value;
+            } elseif ($correctAlias && $object->hasField($field)) {
+                $tableOrder[$tableAlias . '.' . $field] = $value;
+            } elseif (!$correctAlias && $whitelisted) {
+                $tableOrder[$alias . '.' . $field] = $value;
+            }
+        }
+
+        return $tableOrder;
+    }
+
+    /**
+     * Check the limit parameter and ensure it's within the maxLimit bounds.
+     *
+     * @param array $options An array of options with a limit key to be checked.
+     * @return array An array of options for pagination
```

Comment:
It would be better to write and submit a sniffer for this instead of adding noise comments for this very trivial thing 10+ times in a PR, cloaking the more relevant issues at hand.